src/core/core/task.cpp

```diff
@@ -182,7 +183,7 @@ void task::exec_internal()
             else
             {
                 _state.compare_exchange_strong(READY_STATE, TASK_STATE_CANCELLED);
-                _spec->on_task_end.execute(this);
+                _spec->on_task_cancelled.execute(this);
```

Comment:
for the current exec(), this is still on_task_end. on_task_cancelled is executed in task::cancel